helix-core/src/main/java/org/apache/helix/manager/zk/GenericBaseDataAccessorBuilder.java

```diff
@@ -123,23 +124,26 @@ protected RealmAwareZkClient createZkClient(RealmAwareZkClient.RealmMode realmMo
 
   /**
    * Validate ZkClientType based on RealmMode.
+   * If ZkClientType is DEDICATED or SHARED, the realm mode must be SINGLE-REALM.
+   * If ZkClientType is FEDERATED, the realm mode must be MULTI-REALM.
    * @param zkClientType
    * @param realmMode
    */
   private void validateZkClientType(ZkClientType zkClientType,
       RealmAwareZkClient.RealmMode realmMode) {
-    boolean isZkClientTypeSet = zkClientType != null;
-    // If ZkClientType is set, RealmMode must either be single-realm or not set.
-    if (isZkClientTypeSet && realmMode == RealmAwareZkClient.RealmMode.MULTI_REALM) {
-      throw new HelixException("ZkClientType cannot be set on multi-realm mode!");
+    if (realmMode == null) {
```

Comment:
Do we have some default mode for it? Shall we make it default behavior as single realm?